internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -425,6 +508,22 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 		}
 	}
 
+	// Patch composite claim - ClaimConfigurator is expected to properly
+	// handle nil composite claims if claim is not set for this composite
+	if err := r.composite.PatchClaim(ctx, cc, client.MergeFrom(ccActual)); err != nil {
+		log.Debug(errUpdateClaim, "error", err)
+		r.record.Event(cr, event.Warning(reasonCompose, err))
+		return reconcile.Result{RequeueAfter: shortWait}, nil
+	}
+
+	// Update composite with resource reference changes and potential
+	// patches from composed resource
+	if err := r.client.Patch(ctx, cr, client.MergeFrom(crActual)); err != nil {
+		log.Debug(errUpdate, "error", err)
+		r.record.Event(cr, event.Warning(reasonCompose, err))
+		return reconcile.Result{RequeueAfter: shortWait}, nil
+	}
```

Comment:
Since the reconcile already performs a Get on the composite, it seemed like a patch with a merge diff offered a smaller patch to reduce field conflict between composite and claim reconciler.

Reply:
Got it. Though it occurs to me now that we were just calling `Update` before, since we should keep a relatively fresh view on the composite throughout the reconcile. Did you go with Patch to avoid resource version conflicts?